packages/babel-helper-hoist-variables/tsconfig.json

```diff
@@ -0,0 +1,14 @@
+{
+  "generated": false,
+  "extends": "../../tsconfig.base.json",
+  "compilerOptions": {
+    "outDir": "./lib",
+    "rootDir": "./src"
+  },
+  "include": ["./src/**/*"],
+  "references": [
+    {
+      "path": "../babel-types"
```

Comment:
Currently the [tsconfig generator](https://github.com/babel/babel/blob/672a58660f0b15691c44582f1f3fdcdac0fa0d2f/scripts/generators/tsconfig.js#L50) will generate one more item:
```json
{ "path": "../babel-traverse" }
```

I am not familiar with the `references` usage here but the type checking and the VSCode type inferences seems to work after I removed `babel-traverse`, which causes cycling dependencies since this package is also depended by `@babel/traverse`.

Should we filter out dependencies from introduced only from type imports? In this case we have `import type { NodePath } from "@babel/traverse"`.

/cc @nicolo-ribaudo @zxbodya

Reply:
The dependency is needed also for type imports: it's needed to tell TS in which order to compile the packages to properly type-check everything.

TS doesn't support circular dependencies yet (https://github.com/microsoft/TypeScript/issues/33685), so we could consider moving to type-checking all the packages as a whole rather than package by package (it will make incremental type-checks a bit slower, but it doesn't have this ordering problem).

Reply:
I would suggest, if removing reference to `@babel/traverse` we should also remove type import:

`import type { NodePath } from "@babel/traverse";`

Might be using `any` for now.

Otherwise it might result in some weird errors - something like "can not output to file used as input" (basically compiled TS would be imported here, but when compiling `@babel/traverse` this file would be used as input, which would mean - when compiling `@babel/traverse` TS will try to use output of it as an import )